### PR TITLE
Create express routing and index.js files

### LINKED ISSUE #30: Add an index.js file
Can someone PR an index file that says "Hello world"?

### LINKED ISSUE #38: Add a routing file that uses express
It's possible we can use a starter template for this, too. See what looks good to you?

### PR DESCRIPTION
**Pull request by VernL**
This PR closes #30 & #38.

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (RichardLitt) [inline src/package.json]: Isn't there already a package.json? Why another one?

File: src/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
+var router = express.Router();
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  res.render('index', { title: 'Express' });
```
Comment (RichardLitt) [inline src/routes/index.js]: Is this the title you want?

File: src/routes/users.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
+var router = express.Router();
+
+/* GET users listing. */
+router.get('/', function(req, res, next) {
```
Comment (RichardLitt) [inline src/routes/users.js]: Why is there a users route?

File: src/views/error.ejs [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<h1><%= message %></h1>
```
Comment (RichardLitt) [inline src/views/error.ejs]: What templating language is this? Why this one?

**Comment by VernL** [comment]
I believe my initial PR went outside the scope of issue #39 . I have updated the PR to better reflect the issue at hand.

File: src/views/error.ejs (hunk 4 above)
Comment (VernL) [reply src/views/error.ejs]: Not a great answer, but it's the one I am familiar with. It was quick to learn as you write regular HTML and wrap your JS with <% %>.

**Comment by RichardLitt** [comment]
Thank you! Scoping is an awesome thing to learn. 

You need to commit the package-lock.json file. As well, there's no .gitignore. Can you add one for `node_modules`?

**Comment by VernL** [comment]
There we go, added a nice gitignore file. 

> You need to commit the package-lock.json file.

I believe the package-lock.json file was part of my original commit. Unless I am missing something, it should be up to date.

**Comment by RichardLitt** [comment]
It isn't. Try rimraffing node_modules, running `npm i` again, and then commit the new package-lock.json file.

**Comment by RichardLitt** [comment]
Also - rebase on master! :D

**Comment by RichardLitt** [comment]
@VernL Let's work on branches on this repo, instead of on your own private fork, in the future. I want @hammerhead79 to be able to work on this branch, too, but he can't push to your fork. That OK?

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,249 @@
+{
```
Comment (VernL) [inline package-lock.json]: Is there a problem with this file??

Comment (RichardLitt) [reply package-lock.json]: It didn't match the one I had on my machine (was that expected? I don't think so).

**Comment by RichardLitt** [comment]
Can you rebase?

**Comment by VernL** [comment]
> Can you rebase? 

Done

**Comment by RichardLitt** [comment]
Doesn't look like it. I still have a conflict. :) 

screen shot 2017-10-09 at 1 31 19 pm

**Comment by VernL** [comment]
There is a merge conflict with the index.js file in the master branch. Should I resolve and merge?

**Comment by RichardLitt** [comment]
Nope! Don't merge - use `git rebase master` and resolve it locally. You should be able to rebase on master, and then use force push to your branch on this PR so that there are no more conflicts on this branch. Rebasing is an awesome tool - if you're not sure how it works, there are definitely some guides you can find.

**Comment by VernL** [comment]
In case anyone else needs to rebase from another branch:  
- Create a remote that points to the upstream repository ([like this](https://help.github.com/articles/configuring-a-remote-for-a-fork/))
- use `git remote -v` to see connected remotes
- run `fetch <remote_name>` This is needed before you can work with a remote! 
- from your branch, run `git rebase --interactive <other_branch_name>` https://help.github.com/articles/about-git-rebase/

**Comment by RichardLitt** [comment]
I normally use `git fetch --all` - makes it easier. `--interactive` also isn't needed, though it makes it a bit easier sometimes. :)

**Comment by RichardLitt** [comment]
Works for me! Thank you!

**Pull request merged**